src/odemis/acq/stream_settings.py

```diff
@@ -0,0 +1,174 @@
+        self._read()  # update the config_data by reading the JSON file
+        self.settings_order = None  # the ordered list of setting parameters used for loading any given stream
+
+    def _read(self):
+        logging.debug("Reading the most recently used stream settings in JSON")
+        if os.path.exists(self.file_path):
+            try:
+                with open(self.file_path, 'r') as jsonfile:
+                    self.config_data = json.load(jsonfile)
+            except json.decoder.JSONDecodeError as e:
+                logging.warning(f"Error decoding JSON: {e}")
+            except Exception as e:
+                logging.warning(f"Error in reading the stream settings: {e}")
+
+    def _write(self):
+        logging.debug("Writing the most recently used stream settings in JSON")
+        try:
+            if self.config_data:
+                with open(self.file_path, 'w') as jsonfile:
+                    json.dump(self.config_data, jsonfile, indent=2)
+        except Exception as e:
+            logging.warning(f"Error in writing the stream settings: {e}")
+
+    def update_data(self, new_entries: List[Dict[str, Any]]):
+        """
+        Update the saved stream setting with the list of new_entries. First element in JSON file
+        is the latest entry while the last element is the oldest stream setting. The file updates
+        its entries based on the new_entries while maintaining a fixed length of entries in the file.
+        :param new_entries: (list of stream controller settings) Each element in the list represent the setting
+         of the stream saved as a list from oldest to latest stream setting
+        """
+        # Convert config data to list to enable list operations
+        if self.config_data is None:
+            self.config_data = []
+        data = self.config_data
+
+        for e in new_entries:
+            # find if the new entry name exists in the saved data
+            index = get_config_index(data, e["name"])
+            # pop the entry from the list if the stream name is same to avoid duplication
+            if index is not None:
+                data.pop(index)
+            # insert the new entry at the first position
+            data.insert(0, e)
+            # length of list of entries should be constant
+            data = data[:self.max_entries]
+
+        self.config_data = data
+        self._write()
+
+    def get_settings_order(self, stream: FluoStream):
+        """
+        Get the VAs for loading a stream setting and sort them so it can applied in the given order while loading
+        a stream later.
+        :param stream: stream object that consists of VAs
+        """
+        if self.settings_order is None:
+            stream_vas = model.getVAs(stream)
+            settings = set(stream_vas.keys()).difference(NON_SETTINGS_VA)
+            self.settings_order = util.sorted_according_to(settings, SETTINGS_ORDER)
+
+    def get_streams_settings(self, streams: list):
```

Comment:
The method names `get_stream_settings()`/`set_stream_settings()` are confusing. Especially the `get` doesn't return any value, but instead write something in a file! The `set` reads a file and modifies a stream based on that content. 

I think you have two options. The most straightforward is to keep these 2 methods, but adjust the names. I'd suggest something like `update_entries(streams)` and `apply_settings(stream, entry)`.

A second option would be to separate the notion of reading/writing the config file from the conversion stream<>settings (that's what I had in mind originally in my first review).
So in this class, there would be just two methods `update_data(settings)` (already there, above) and `get_entry(name) -> Dict[str, Any]` (which just returns the settings for the requested entry.
Separately, as simple functions of the module, there would be a `extract_settings_from_stream(stream)` and `apply_settings_to_stream(settings, stream)`.

I like the second option more because it feels like there is a better separation of concerns. So maybe more easily reused. However, the first approach might just be more straightforward when using it now. So I'm fine with whichever option.